system/Pager/Pager.php

```diff
@@ -154,8 +154,7 @@ protected function displayLinks(string $group, string $template): string
 			throw PagerException::forInvalidTemplate($template);
 		}
 
-		return $this->view->setVar('pager', $pager)
-						->render($this->config->templates[$template]);
+		return view($this->config->templates[$template], ['pager' => $pager]);
```

Comment:
Why remove the call to the local instance? I think this needs to use the injected renderer.

Reply:
I've tested the code using the view() method and it works properly. The issue with local rendered is that you cannot that easily override the path of the view. 

What I'm doing to load different template is that I prepend the template namespace to the path for example

'default_full'   => 'CodeIgniter\Pager\Views\default_full',

I end up loading the view

\Templates\{ConfiguredTemplateName}\CodeIgniter\Pager\Views\default_full 

And I'm doing the prepend inside the view() method. 

I don't see why we need to use the injected view renderer. 

view is passed as AppServices::renderer() in the pager service;
common view() method will call $renderer = Services::renderer(); 

The issue here is that in order to modify the view path for pager with the current setup it is needed to either create a custom rendered class or override the display method with custom pager class. Having it call view() method makes it a lot easier to achieve this as you'll need to override a single method only. IMO RendererInterface should be completely removed from pager and have it rely on the view() method

Reply:
There's nothing magic about `view()`, it just loads and configures the Renderer Service then calls `setData()`: https://github.com/codeigniter4/CodeIgniter4/blob/4ac34c323cd6f074719a0ac713fb31038dec6448/system/Common.php#L1234-L1250

We already have an instance of `RenderInterface` injected so we need to use that. There is no guarantee that this is the same Renderer that will be accessed by `view()` so making this switch introduces a potential breaking change. I am not disagreeing with your point that the dependency *should not* be this way - just that it is and we're stuck with it for now regardless.